lib/url_formatter.rb

```diff
@@ -19,26 +19,34 @@ def minimal (input = url)
     uri = Addressable::URI.parse(input)
     queries = CGI::parse(uri.query) unless uri.query.nil?
     if queries.nil?
-      return input.gsub(/(\?|#).*$/, '')
+      input.gsub(/[?#].*$/, "")
     else
-      queries.keep_if { |k,v| ['sid'].include? k }
-      querystring = ('?' + URI.encode_www_form(queries)) unless queries.empty?
-      return input.gsub(/(\?|#).*$/, '') << querystring.to_s
+      queries.keep_if { |k, _| ["sid"].include? k }
+      querystring = "?#{URI.encode_www_form(queries)}" unless queries.empty?
+      input.gsub(/[?#].*$/, "") << querystring.to_s
     end
   end
 
-  def minimal_no_http
-    minimal.gsub(/https?:\/\/www\./, "")
+  def minimal_no_protocol_no_www
+    minimal.gsub(%r{https?://(www\.)?}, "")
   end
   
   def no_www
-    minimal.gsub(/http:\/\/www\./, "http://")
+    minimal.gsub(%r{(https?)://www\.}, "\\1://")
   end
   
   def with_www
-    minimal.gsub(/http:\/\//, "http://www.")
+    minimal.gsub(%r{(https?)://}, "\\1://www.")
   end
-  
+
+  def with_http
+    minimal.gsub(%r{https?://}, "").prepend("http://")
```

Comment:
would this be better as     minimal.gsub(%r{^https?://}, "http://")

Reply:
That does make more sense!